test/jellyfish_web/controllers/component_controller_test.exs

```diff
@@ -111,6 +114,76 @@ defmodule JellyfishWeb.ComponentControllerTest do
       assert {:ok, _removed_files} = room_id |> HLS.Recording.directory() |> File.rm_rf()
     end
 
+    setup :set_mox_from_context
+    setup :verify_on_exit!
+
+    test "renders component with s3 credentials", %{conn: conn} do
+      conn = post(conn, ~p"/room", videoCodec: "h264")
+      assert %{"id" => room_id} = json_response(conn, :created)["data"]["room"]
+
+      bucket = "bucket"
+
+      conn =
+        post(conn, ~p"/room/#{room_id}/component",
+          type: "hls",
+          options: %{
+            persistent: false,
+            s3: %{
+              accessKeyId: "access_key_id",
+              secretAccessKey: "secret_access_key",
+              region: "region",
+              bucket: bucket
+            }
+          }
+        )
+
+      assert response =
+               %{
+                 "data" => %{
+                   "type" => "hls",
+                   "metadata" => %{
+                     "playable" => false,
+                     "lowLatency" => false,
+                     "persistent" => false,
+                     "targetWindowDuration" => nil
+                   }
+                 }
+               } =
+               json_response(conn, :created)
+
+      parent = self()
+      ref = make_ref()
+
+      expect(ExAws.Request.HttpMock, :request, 4, fn _method,
+                                                     url,
+                                                     req_body,
+                                                     _headers,
+                                                     _http_opts ->
+        assert req_body == @body
+
+        assert String.contains?(url, bucket)
+
+        assert url |> String.split("/") |> List.last() |> then(&Enum.member?(@files, &1))
```

Comment:
Shouldn't we check the equality of this URL with string interpolation? This way, we would have one assert.

Reply:
I forgot that `String.ends_with?/2` accept list